### PR TITLE
Fixes #84 - Offset bytes in StreamFileTypeProvider

### LINKED ISSUE #84: Magic Bytes offset on Streams
Thanks for this library, I've found it very useful in a project I'm working on. However I've found that when evaluating magic bytes with the `StreamFileTypeProvider`, the offset is never considered, meaning that the `mp4` format will never be discovered (or others that might be supported in the future that have offset bytes)

I'm sending a pull request with this fix and an optimization when ordering the sizes to find the maximum bytes size for the buffer,

Thanks for your hard work!

### PR DESCRIPTION
**Pull request by ciosoriog**
Fix #84

- Fix evaluating magic bytes with the `StreamFileTypeProvider`, offset consideration
- Optimized sorting by sizes for max size of buffer with offset and access speed (don't look for `.Last()`, instead since you are already sorting, sort by descending length and grab the `.First()`, faster memory access

### CODE REVIEW

File: MagicBytesValidator/Services/Streams/StreamFileTypeProvider.cs [hunk 1]
```diff
@@ -29,23 +29,19 @@ CancellationToken cancellationToken
         var sequencesToFileTypes = _mapping.FileTypes
             .Select(fileType =>
                 fileType.MagicByteSequences
-                    .Select(sequence => (sequence, fileType))
+                    .Select(sequence => (length: sequence.Length + fileType.MagicByteOffset, sequence, fileType))
```
Comment (kryptobi) [inline MagicBytesValidator/Services/Streams/StreamFileTypeProvider.cs:32]:
```suggestion
                    .Select(sequence => (sequence.Length + fileType.MagicByteOffset, sequence, fileType))
```

Identifiers are not necessary

**Review by kryptobi — APPROVED** [review]

**Comment by kryptobi** [comment]
> lgtm. If @kryptobi approves too, you're free to merge :)

@ciosoriog feel free to merge it.

File: MagicBytesValidator/Services/Streams/StreamFileTypeProvider.cs (hunk 1 above)
Comment (ciosoriog) [reply MagicBytesValidator/Services/Streams/StreamFileTypeProvider.cs:32]: I agree, they aren't necessary but makes for a more readable code than, say using `Item1` in line 35, but if you want me to remove it, I have no problem with it 😃

**Comment by ciosoriog** [comment]
> > lgtm. If @kryptobi approves too, you're free to merge :)
> 
> @ciosoriog feel free to merge it.

Thank you @kryptobi, but I don't have write access to merge the PR :)

**Comment by kryptobi** [comment]
> > > lgtm. If @kryptobi approves too, you're free to merge :)
> 
> > 
> 
> > @ciosoriog feel free to merge it.
> 
> 
> 
> Thank you @kryptobi, but I don't have write access to merge the PR :) 

Oh ok. I did it. Thanks a lot! I appriciate it

**Pull request merged**